### ISSUE TITLE
Extensión de Context.java

### ISSUE THREAD

**Issue by Isabel-Roman** [MIT-FS/Audit4Improve-API#31]
Implementar los métodos de Context.java que estaban pendientes
Añadir la gestión de las propiedades de fuente en el fichero de configuración para Indicadores por estado:
Indicator.critical/warning/ok.
font.color/height/type
**ATENCIÓN:** Si existe configuración personalizada del cliente (appConfFile) sobreescribe a la establecida en el fichero por defecto (a4i.conf)
Comprobar antes a4i.conf
Implementar el método
public static Font getIndicatorFont(Indicator.State state) throws IOException
Revisar el modelo:

![configPackageV 0 2](https://user-images.githubusercontent.com/3404275/165087325-f2c69b5e-51be-47c7-a04a-f572c13b5dde.GIF)
Repartir métodos a implementar y tests en la pareja, ambos deben codificar tanto tests como código

**Comment by jls98** [comment]
@Isabel-Roman 
No entiendo la tarea. Tenemos que finalizar los métodos getDefaultFont, getMetricFont y getIndicatorFont? Además, no entiendo que tenemos que hacer con los objectos de Font. Hay algunos requisitos más?

**Comment by Isabel-Roman** [comment]
Sí, hay que implementar los métodos que aún no lo estaban. El método getIndicatorFont devuelve la fuente para el estado del indicador que se pase como parámetro... para eso en el fichero de configuración hay que distinguir el valor de la fuente en función del estado del indicador.
No hay más requisitos, pero si tenéis dudas podéis pedir tutoría.
Recordad que tenéis que repartir el trabajo entre los dos

**Comment by elimarval** [comment]
@Isabel-Roman estamos intentando implementar los métodos que devuelven Font y nos encontramos que de algún modo u otro hay que completar un atributo color. En cambio por más que hemos buscado en las referencias de Java no existe el atributo color para la clase Font. Por lo que no podemos rellenar esa característica y sólo nos quedaríamos con el tipo de letra y el tamaño. Al parecer el color solo se puede aplicar una vez se crea un texto con esa fuente, por ejemplo:

JTextArea txt = new JTextArea();
Font font = new Font("Verdana", Font.BOLD, 12);
txt.setFont(font);
txt.setForeground(Color.BLUE);

(el código es de stack overflow).

También tenemos la duda de si para los indicadores WARNING y CRITICAL tenemos que inventarnos la fuente correspondiente o tienes una especificación en concreto de cómo tienen que ser las fuentes para esos estados.

**Comment by Isabel-Roman** [comment]
Font no tiene que ser una clase de la librería Java. Podríais usarla, pero si no contiene todo lo necesario hay que crear una nueva
Podéis reutilizar Cualquier clase de Java, pero lo que se necesite y no se tenga hay que añadirlo.
Para eso vuestra clase Font puede contener un Font de java, o heredar, lo que veáis más apropiado
Aquí el requisito es:
El cliente puede elegir el color con el que se muestran los indicadores según el estado
Así que el fichero de configuración debe permitir indicar el color por estado (por cada estado) y el código debe permitir, leerlo y aplicar el color adecuado cuando se vaya a persistir el informe
Un saludo
